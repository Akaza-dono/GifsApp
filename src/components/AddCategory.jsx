import { useState } from "react"

export const AddCategory = ({onNewValue}) => {
    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) =>{
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return
        // onAddCategory(e => [inputValue, ...e])
        onNewValue(inputValue.trim())
        setInputValue('')
    }

    return(
        <form onSubmit={ e => onSubmit(e) }>
            <input 
                type="text"  
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}