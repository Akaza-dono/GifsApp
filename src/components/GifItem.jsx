import  PropTypes  from "prop-types"

export const GifItem = ({title,url}) => {
    return(
        <div className="card">
            <img src={url} alt={title} />
            <p name="parra">{title}</p>
        </div>
    )
}

GifItem.propTypes = { 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}