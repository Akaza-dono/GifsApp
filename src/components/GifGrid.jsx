import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from 'prop-types'
// import '../../src/styles.css';

export const GifGrid = ({category}) => {
    const {images, isLoading} = useFetchGifs(category);

    return(
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem key={image.id} { ...image }/>
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

