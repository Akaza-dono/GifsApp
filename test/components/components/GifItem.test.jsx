import { render, screen } from "@testing-library/react"
import { GifItem } from "../../../src/components/GifItem";

describe('Suit de pruebas para la aplicaciond de gifs',()=>{
    const title = "Un titulo";
    const url = "unauurl"

    test('El gifItem debe hacer match con el snapshot',()=>{
        const { container } = render(<GifItem title={title} url={url}/>)
        expect( container ).toMatchSnapshot();
   }) 

    // test('Debe de confirmar que las prop types esten resueltas',()=>{
    //     render(<GifItem title={title} url={url}/>)
    //     expect(screen.getByText(title).innerHTML).toBe(title)
    // });

    test('debe de mostrar la imagen con el url y el alt indicado',()=>{
        render(<GifItem title={title} url={url}/>)
        const {src, alt} = screen.getByRole('img');
        expect(src.split('/')[3]).toBe(url);
    });

    test('Debe de reconocer que actualmente el titulo exite',()=>{
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title).innerHTML).toBeTruthy();
    })

})