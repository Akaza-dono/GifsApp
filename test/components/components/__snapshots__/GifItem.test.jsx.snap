// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suit de pruebas para la aplicaciond de gifs El gifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo"
      src="unauurl"
    />
    <p
      name="parra"
    >
      Un titulo
    </p>
  </div>
</div>
`;
